sqrl-planner/sqrl-planner-local/src/main/java/com/datasqrl/sql/FlinkSqlGenerator.java

```diff
@@ -0,0 +1,453 @@
+      SinkFactory sinkFactory = (SinkFactory) connectorFactory;
+      Object o = sinkFactory.create(
+          new FlinkSinkFactoryContext(name, tableConfig, formatFactory));
+      if (o instanceof DataStream) {
+        throw new RuntimeException("Not supported");
+      } else if (o instanceof TableDescriptor.Builder) {
+        TableDescriptor.Builder builder = (TableDescriptor.Builder) o;
+        TableDescriptor descriptor = builder.schema(toSchema(table.getSchema()))
+            .build();
+
+        return toCreateTable(name, descriptor, (RelDataType) table.getRelDataType(),
+            (Optional<SqlNode>) table.getWatermarkExpression(),
+            (Optional<SqlNode>) table.getWatermarkColumn(), Optional.empty(),
+            false, table.getSchema().getPrimaryKey());
+      } else {
+        throw new RuntimeException("Unknown sink type");
+      }
+
+    } else {
+      throw new RuntimeException("Unknown factory type " + connectorFactory.getClass().getName());
+    }
+  }
+
+  private SqlCreateTable toCreateTable(String name, TableDescriptor descriptor, RelDataType relDataType,
+      Optional<SqlNode> watermarkExpression, Optional<SqlNode> watermarkColumn,
+      Optional<String> sourceTimeMetaData, boolean setMetadata, List<String> primaryKey) {
+    return new SqlCreateTable(SqlParserPos.ZERO,
+        identifier(name),
+        createColumns(descriptor.getSchema(), relDataType, watermarkExpression, watermarkColumn, sourceTimeMetaData, setMetadata), //columns
+        createConstraints(descriptor.getSchema(), primaryKey), //table constraints
+        createProperties(descriptor.getOptions()), //property list ?
+        createPartitionKeys(descriptor.getPartitionKeys()), //partition key list
+        createWatermark(descriptor.getSchema(),watermarkColumn, watermarkExpression, setMetadata),
+        createComment(descriptor.getComment()),
+        true,
+        false
+    );
+  }
+
+  private SqlWatermark createWatermark(Optional<Schema> schema,
+      Optional<SqlNode> watermarkColumn, Optional<SqlNode> watermarkExpression,
+      boolean setMetadata) {
+    if (!setMetadata) {
+      return null;
+    }
+    final String watermarkName;
+    final WaterMarkType waterMarkType;
+    if (watermarkColumn.isPresent()) { //watermark is a timestamp column
+      watermarkName = removeAllQuotes(RelToFlinkSql.convertToString(watermarkColumn.get()));
+      if (ReservedName.SOURCE_TIME.matches(watermarkName)) {
+        waterMarkType = WaterMarkType.SOURCE_WATERMARK;
+      } else {
+        waterMarkType = WaterMarkType.COLUMN_BY_NAME;
+      }
+    } else { //watermark is a timestamp expression
+      Preconditions.checkArgument(watermarkExpression.isPresent());
+      SqlCall call = (SqlCall) watermarkExpression.get();
+
+      SqlNode expr = stripAs(call);
+
+      if (expr instanceof SqlIdentifier && ReservedName.SOURCE_TIME.matches(((SqlIdentifier)expr).getSimple())) {
+        waterMarkType = WaterMarkType.SOURCE_WATERMARK;
```

Comment:
Everything is a table source now so `WaterMarkType.SOURCE_WATERMARK` no longer applies and all these if-else statements go away.